### PR TITLE
Re-enable test_protect_kms_transform

### LINKED ISSUE #316: Reenable test_protect_kms_transform
We had to disable this in https://github.com/shotover/shotover-proxy/pull/315 due to making the repo public preventing the use of github secrets.

Maybe we can mock out the AWS API like https://github.com/spulec/moto
Or maybe theres a way to safely distribute the key or a lower access key.

### PR DESCRIPTION
**Pull request by Claude-at-Instaclustr**
Adds moto as an AWS mock

fixes #316

### CODE REVIEW

File: shotover-proxy/src/transforms/protect/mod.rs [hunk 1]
```diff
@@ -421,11 +422,38 @@ mod protect_transform_tests {
         panic!()
     }
 
-    //#[tokio::test(flavor = "multi_thread")]
+    /// Creates a new AWS key and returns the key ARN
+    async fn create_aws_key() -> Result<String> {
+        let client = reqwest::Client::new();
+
+        let res = client.post("http://localhost:5000")
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/mod.rs:429]: Can use a [raw string literal](https://doc.rust-lang.org/reference/tokens.html#raw-string-literals) here

File: shotover-proxy/src/transforms/protect/mod.rs [hunk 2]
```diff
@@ -421,11 +422,38 @@ mod protect_transform_tests {
...
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
     //#[ignore] // reason: requires AWS credentials
-    #[allow(unused)]
+    //#[allow(unused)]
     // Had to disable this when the repo went public as we dont have access to github secrets anymore
     async fn test_protect_kms_transform() -> Result<()> {
+        let _compose = DockerCompose::new("tests/transforms/docker-compose-moto.yml")
+            .wait_for("Press CTRL\\+C to quit");
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/mod.rs:453]: Can use raw string literal here too.

File: shotover-proxy/src/transforms/protect/key_management.rs [hunk 3]
```diff
@@ -48,8 +49,18 @@ impl KeyManagerConfig {
...
+                endpoint,
             } => Ok(KeyManager::AWSKms(AWSKeyManagement {
-                client: KmsClient::new(Region::from_str(region.as_str())?),
+                client: {
+                    let region_obj = match endpoint {
+                        Some(x) => Region::Custom {
+                            name: Region::from_str(&*region)?.name().to_string(),
+                            endpoint: x,
+                        },
+                        _ => Region::from_str(&*region)?,
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/key_management.rs:60]: Borrow and deference? We only need the borrow.

File: shotover-proxy/src/transforms/protect/mod.rs [hunk 4]
```diff
@@ -421,11 +422,38 @@ mod protect_transform_tests {
...
+        Ok(parsed["KeyMetadata"]["Arn"].as_str().unwrap().to_string())
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
     //#[ignore] // reason: requires AWS credentials
-    #[allow(unused)]
+    //#[allow(unused)]
     // Had to disable this when the repo went public as we dont have access to github secrets anymore
     async fn test_protect_kms_transform() -> Result<()> {
+        let _compose = DockerCompose::new("tests/transforms/docker-compose-moto.yml")
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/mod.rs:452]: This is an integration test, so it should go with the other integration tests. You would typically only put unit tests at the bottom of a file.

File: shotover-proxy/src/transforms/protect/mod.rs [hunk 5]
```diff
@@ -421,11 +422,38 @@ mod protect_transform_tests {
...
+
+        let result = res.text().await?;
+        let parsed: serde_json::Value = serde_json::from_str(result.as_str())?;
+        Ok(parsed["KeyMetadata"]["Arn"].as_str().unwrap().to_string())
+    }
+
+    #[tokio::test(flavor = "multi_thread")]
     //#[ignore] // reason: requires AWS credentials
-    #[allow(unused)]
+    //#[allow(unused)]
```
Comment (conorbros) [inline shotover-proxy/src/transforms/protect/mod.rs:449]: Should delete this commented out stuff if we don't need it anymore.

**Review by conorbros — CHANGES_REQUESTED** [review]

**Comment by conorbros** [comment]
Looks like you need to run `cargo fmt`

File: shotover-proxy/src/transforms/protect/mod.rs (hunk 4 above)
Comment (Claude-at-Instaclustr) [reply shotover-proxy/src/transforms/protect/mod.rs:452]: I did not place the test, just fixed it.

File: shotover-proxy/src/transforms/protect/key_management.rs (hunk 3 above)
Comment (Claude-at-Instaclustr) [reply shotover-proxy/src/transforms/protect/key_management.rs]: done

File: shotover-proxy/src/transforms/protect/mod.rs (hunk 5 above)
Comment (Claude-at-Instaclustr) [reply shotover-proxy/src/transforms/protect/mod.rs]: done

File: shotover-proxy/src/transforms/protect/mod.rs (hunk 4 above)
Comment (rukai) [reply shotover-proxy/src/transforms/protect/mod.rs:449]: Yeah I wouldnt call this an integration test unless we actually run full shotover against a cassandra instance like our other integration tests do.
Also its already here, moving it in this PR would mess up the diff.

**Review by rukai — APPROVED** [review]
If rusoto has an API for creating the key we should use that instead, but I dont consider that a blocker because we will want to replace rusoto with amazons official crate at some point.

Nice work getting this test running again!

File: shotover-proxy/src/transforms/protect/mod.rs (hunk 4 above)
Comment (conorbros) [reply shotover-proxy/src/transforms/protect/mod.rs:449]: Yep, I was incorrect. The `DockerCompose` threw me off.

**Review by conorbros — APPROVED** [review]
LGTM!

**Pull request merged**